security/src/main/java/io/micronaut/security/authentication/Authenticator.java

```diff
@@ -174,8 +181,17 @@ private AuthenticationResponse authenticateAll(@NonNull T requestContext,
     private List<ReactiveAuthenticationProvider<T, I, S>> everyProviderSorted() {
         List<ReactiveAuthenticationProvider<T, I, S>> providers = new ArrayList<>(reactiveAuthenticationProviders);
         if (beanContext != null) {
+
             providers.addAll(imperativeAuthenticationProviders.stream()
-                    .map(imperativeAuthenticationProvider -> new AuthenticationProviderAdapter<>(beanContext, scheduler, imperativeAuthenticationProvider)).toList());
+                    .map(imperativeAuthenticationProvider -> {
+                        if (imperativeAuthenticationProvider instanceof ExecutorAuthenticationProvider<?, ?, ?> ap && beanContext.containsBean(ExecutorService.class, Qualifiers.byName(ap.getExecutorName()))) {
```

Comment:
You can probably cache the bean context invocations

Reply:
something like https://github.com/micronaut-projects/micronaut-security/pull/1548/commits/08fca1baac9c1e1e9a13b779c912186e842d0449

Reply:
Yes